### DISCUSSION TITLE
adding file during the osbuild phase

### DISCUSSION THREAD

**Discussion by lucamaf** [ostreedev/ostree#3248] [Q&A]
Not sure if this is the correct place but I'm wondering how to add a big file (imagine a DB backup or VM image) in the ostree filesystem at build time. 
Reason behind this is imagine having a partially disconnected Edge Device which requires these type of artifacts to boot and there is no easy way to transfer them after installation time.
I tried already in some way manipulating kickstart or creating custom rpm to include those files but none of them worked.
I would ask this as an enhancement, unless there is an easier way to do that with ostree.

**Comment by ericcurtin** [comment]
Highly recommend creating an RPM and setting up an RPM repo to pull from as you suggested that you tried and failed.

There are other ways, but it's probably the easiest for you.

**Comment by ericcurtin** [comment]
I'd point you at our Automotive tools, but they are bespoke right now to Automotive.

Long term this is just a Containerfile/Dockerfile btw with bootc, so you can do whatever in a RUN command.

**Reply by lucamaf** [reply]
that would be the easiest approach once we have feature parity between bootc and imagebuilder

**Comment by ericcurtin** [comment]
You can try this if you want it's new:

https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/using_image_mode_for_rhel_to_build_deploy_and_manage_operating_systems/introducing-image-mode-for-rhel_using-image-mode-for-rhel-to-build-deploy-and-manage-operating-systems

can also be based on Fedora bootc if something recent is needed

**Comment by cgwalters** [comment]
You don't want it in the "ostree filesystem", you want it in "the filesystem" specifically in `/var` right?  IOW it's not lifecycled with the commit/image, it's extra data.

Another simple example of this is "floating" container images, flatpaks etc.  At the current time ostree doesn't ship anything opinionated for this, it's the job of an external provisioning system which can mount `/var` and write whatever it wants at that time.  (e.g. anaconda or osbuild or whatever)

However on the bootc side we are likely to introduce "data containers" for this.

**Reply by lucamaf** [reply]
agreed on your first paragraph, when I try to add content using kickstart though I cannot find it in the installed system. This is what I have in one of the %post sections

cp -f /run/install/repo/files/test.txt /var

and cannot see anything in /var of the installed system
